draft-packages/menu/KaizenDraft/Menu/Menu.tsx

```diff
@@ -1,84 +1,90 @@
 import { ButtonProps } from "@kaizen/draft-button"
-import * as React from "react"
+import React, { useState } from "react"
 import MenuDropdown from "./MenuDropdown"
 
 const styles = require("./styles.scss")
 
-type MenuState = {
-  isMenuVisible: boolean
-}
-
-export type MenuProps = {
+export type GenericMenuProps = {
   align?: "left" | "right"
-  button: React.ReactElement<ButtonProps>
   menuVisible?: boolean
   automationId?: string
+  children: React.ReactNode
 }
 
-export default class Menu extends React.Component<MenuProps, MenuState> {
-  static displayName = "Menu"
-  static defaultProps = {
-    iconPosition: "start",
-    align: "left",
-  }
+type StatefulMenuProps = {
+  button: React.ReactElement<ButtonProps>
+}
 
-  dropdownButtonContainer = React.createRef<HTMLDivElement>()
+export type MenuProps = GenericMenuProps & StatefulMenuProps
 
-  constructor(props: MenuProps) {
-    super(props)
+type Menu = React.FunctionComponent<MenuProps>
 
-    this.state = {
-      isMenuVisible: Boolean(props.menuVisible),
-    }
-  }
+const Menu: Menu = props => {
+  const { align = "left", menuVisible = false } = props
 
-  toggleMenu = (e: MouseEvent) => {
-    e.stopPropagation()
+  const dropdownButtonContainer = React.createRef<HTMLDivElement>()
 
-    const currentState = this.state.isMenuVisible
-    this.setState({
-      isMenuVisible: !currentState,
-    })
-  }
+  const [isMenuVisible, setIsMenuVisible] = useState<boolean>(menuVisible)
```

Comment:
Nice to see hooks here – most of this code was ported from the very old Dropdown and needed some refreshment.